Initial implementation for #405.

knirirr opened the pull request:
I've created a very simple component that displays the number of hits and the page, according to the requirements in #405.

knirirr committed:
Initial implementation for #405.

knirirr committed:
Autogenerated documentation files

terazus reviewed (changes requested):
Add some margin to your component so it's not so crumbled with the pagination bar (you can use class="mt-x" where x is a number).
Also, the number calculation is wrong:
![image](https://user-images.githubusercontent.com/6538906/92400735-3ffbdd00-f0e1-11ea-96b3-981dc01912bc.png)
Here we have 30 to 59 on page 1 where it should be 1 to 29.
So 
`x: function(){return this.currentPage * (this.perPage-1)}`
and
`y: function(){return (this.currentPage * this.perPage) -1}`

knirirr committed:
Some chages based on request in
https://github.com/FAIRsharing/fairsharing.github.io/pull/441#pullrequestreview-483622722

knirirr committed:
Merge branch 'mt_search_hits_405' of github.com:FAIRsharing/fairsharing.github.io into mt_search_hits_405

knirirr committed:
Added missing semicolons.

terazus commented:
The formula is still wrong: we get 0 to 29 instead of 1 to 30 on page 1. Add 1 to all results.

knirirr committed:
Further modification to x/y computation. #405

knirirr committed:
Autogenerated documentation files

knirirr committed:
Another test added for HitCount.

knirirr committed:
Documentation change.

knirirr committed:
Merge branch 'dev' into mt_search_hits_405

knirirr committed:
npm update.

knirirr committed:
Merge branch 'mt_search_hits_405' of github.com:FAIRsharing/fairsharing.github.io into mt_search_hits_405

knirirr committed:
Autogenerated documentation files

knirirr committed:
Removed npm
https://github.com/FAIRsharing/fairsharing.github.io/pull/441

terazus commented on src/store/records.js:
```diff
@@ -9,6 +9,8 @@ export const mutations = {
         state.records = data['records'];
         state.facets = buildFacets(data["aggregations"]);
         state.totalPages = data["totalPages"];
+        state.perPage = data["perPage"];
```
If you want to do that, you also need to add the state to:
1 - the default state: line 63, you need to declare your two new state as null
2 - optionally a mutation that reset the states back to their default value (null or false). You can either use an existing reset mutation or add in new one. If you add a new one, you will need to also call the mutation in the fetchRecords() actions, right before fetching data.

terazus commented on package.json:
```diff
@@ -25,6 +25,7 @@
     "jquery": "^3.5.1",
     "jsdoc-vuex-plugin": "^1.0.0",
     "lodash": "^4.17.20",
+    "npm": "^6.14.8",
```
Please delete.

knirirr committed:
Added mutation and initial state as per
https://github.com/FAIRsharing/fairsharing.github.io/pull/441#discussion_r486423571

knirirr committed:
Autogenerated documentation files

knirirr committed:
Merge branch 'dev' into mt_search_hits_405

knirirr committed:
Merge branch 'mt_search_hits_405' of github.com:FAIRsharing/fairsharing.github.io into mt_search_hits_405

knirirr committed:
Autogenerated documentation files

knirirr committed:
Merge branch 'dev' into mt_search_hits_405

knirirr committed:
Autogenerated documentation files
